### PR TITLE
Add Thread file system enabling instructions

### PR DESCRIPTION
**Pull request by artokin**

### CODE REVIEW

**Comment by artokin** [comment]
@AnotherButler , @terhei , @karsev, @deepakvenugopal would you please review?

**Comment by artokin** [comment]
That page is for `Configuring the hardware`, it contains sub-GHz configuration that does not apply for Thread. In the first though it feels right place for file system documentation. 

Any suggestions for better location? It is not hard to move that chapter.

File: docs/reference/technology/mesh/quick_start_hw.md [hunk 1]
```diff
@@ -45,6 +45,16 @@ The following table shows which development boards have been tested. It does not
 | Onsemi NCS36510 <span style='background-color: #5f5;'>(internal RF)</span> | | | |
 | NXP KW24D <span style='background-color: #5f5;'>(internal RF)</span> | | <span style='background-color: #f00;'>Yes **(4)**</span> | |
 
+#### File System
+
+Thread network stack can save network configurations settings to file system once:
```
Comment (iriark01) [inline docs/reference/technology/mesh/quick_start_hw.md]: "Once" can be interpreted as one time only.

File: docs/reference/technology/mesh/quick_start_hw.md [hunk 2]
```diff
@@ -45,6 +45,16 @@ The following table shows which development boards have been tested. It does not
...
+
+- You enable the file system as instructed in the [Mbed OS storage documentation](https://os.mbed.com/docs/latest/reference/storage.html).
+- You set the file system root path to the Thread network stack by calling the function `ns_file_system_set_root_path(root-path)`. Set the file system root path before starting the Thread stack to read possible configuration settings in the first power up.
+
+When you have enabled the file system, the Thread stack writes network configuration settings to the file system and reads them in the following startups. The size of the Thread configuration settings is a few thousand bytes.
+
+Depending of the selected file system, the application may need to format the file system before you can use it.
```
Comment (iriark01) [inline docs/reference/technology/mesh/quick_start_hw.md]: On, not of

File: docs/reference/technology/mesh/quick_start_hw.md [hunk 3]
```diff
@@ -45,6 +45,16 @@ The following table shows which development boards have been tested. It does not
 | Onsemi NCS36510 <span style='background-color: #5f5;'>(internal RF)</span> | | | |
 | NXP KW24D <span style='background-color: #5f5;'>(internal RF)</span> | | <span style='background-color: #f00;'>Yes **(4)**</span> | |
 
+#### File System
+
+Thread network stack can save network configurations settings to file system once:
+
+- You enable the file system as instructed in the [Mbed OS storage documentation](https://os.mbed.com/docs/latest/reference/storage.html).
```
Comment (iriark01) [inline docs/reference/technology/mesh/quick_start_hw.md]: Do you have to do these in a particular order? Then it should be 1, 2, not bullets

File: docs/reference/technology/mesh/quick_start_hw.md [hunk 4]
```diff
@@ -45,6 +45,16 @@ The following table shows which development boards have been tested. It does not
...
+- You set the file system root path to the Thread network stack by calling the function `ns_file_system_set_root_path(root-path)`. Set the file system root path before starting the Thread stack to read possible configuration settings in the first power up.
```
Comment (iriark01) [inline docs/reference/technology/mesh/quick_start_hw.md]: "Set the file system root path" can be replaced with "do this"; it's shorter and makes it clearer this is all one action

File: docs/reference/technology/mesh/quick_start_hw.md [hunk 5]
```diff
@@ -45,6 +45,16 @@ The following table shows which development boards have been tested. It does not
...
+
+When you have enabled the file system, the Thread stack writes network configuration settings to the file system and reads them in the following startups. The size of the Thread configuration settings is a few thousand bytes.
```
Comment (iriark01) [inline docs/reference/technology/mesh/quick_start_hw.md]: Why is this here, if the action to enable the file system is above? Shouldn't it be together?

**Comment by AnotherButler** [comment]
@MarceloSalazar @ashok-rao Please review at your convenience

**Comment by MarceloSalazar** [comment]
As an initial version, looks good.
I'd like to see more information on how to use the function `ns_file_system_set_root_path` in practice, but maybe this could be added in the future.
